1864 candidate opt in to detailed notifications

camerojo opened the pull request:
This can be merged now - without revealing the optIn/Out functionality which is commented out.

It will be activate in the next version by another issue.

But please approve this so it can be merged.

camerojo committed:
Merge branch 'Flyways' into 1864-candidate-opt-in-to-detailed-notifications

camerojo committed:
Add in allNotifications field.

Improved method naming: get -> construct

camerojo committed:
Factor chat related code out of CandidateOpportunityServiceImpl into OppNotificationService

camerojo committed:
Put in place logic for managing candidate notifications according to their preference.

camerojo committed:
All for updating of notification preference from admin.

camerojo committed:
Add OptIn entry on candidate portal

Commit:
Update Idea config

Commit:
Auto create candidate note when notification preference changes

Commit:
New way of candidate opting in

Commit:
Clarify OtherInfo vs AdditionalInfo

Commit:
Merge branch 'staging' into 1864-candidate-opt-in-to-detailed-notifications

Commit:
Merge branch 'staging' into 1864-candidate-opt-in-to-detailed-notifications

# Conflicts:
#	server/src/main/java/org/tctalent/server/service/db/impl/CandidateOpportunityServiceImpl.java

sadatmalik commented on server/src/main/java/org/tctalent/server/service/db/impl/OppNotificationServiceImpl.java:
```diff
@@ -0,0 +1,327 @@
+    private final UserService userService;
+
+    @Override
+    public void notifyOppChanges(CandidateOpportunity opp, CandidateOpportunityParams changes) {
```
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 14. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.

Commit:
Order services alphabetically

Commit:
Refactored notifications out of JobServiceImpl

camerojo committed:
Added PostService to allow for future plugging in of link previews

camerojo committed:
Use candidate stage description English translations for candidate posts on stage change.

Comment out extractLinkPreviews in PostServiceImpl - with todo

Commit:
Merge branch 'staging' into 1864-candidate-opt-in-to-detailed-notifications

# Conflicts:
#	server/src/main/java/org/tctalent/server/api/admin/CandidateAdminApi.java
#	server/src/main/java/org/tctalent/server/service/db/CandidateService.java
#	server/src/main/java/org/tctalent/server/service/db/impl/CandidateServiceImpl.java
#	ui/admin-portal/src/app/MockData/MockCandidate.ts
#	ui/admin-portal/src/app/components/candidates/view/view-candidate.component.html
#	ui/admin-portal/src/app/components/candidates/view/view-candidate.component.ts
#	ui/admin-portal/src/app/services/candidate.service.ts
#	ui/candidate-portal/src/app/components/profile/view/view-candidate.component.ts

Commit:
Fix logic for posting stage changes.

Add additional post when candidates accepts a job - and handle case where acceptance stage is skipped.

Commit:
Doc only

Commit:
Complete LinkPreview construction

camerojo committed:
Merge branch 'staging' into 1864-candidate-opt-in-to-detailed-notifications

camerojo committed:
Add meaningful title to Duolingo html coupon email

camerojo committed:
Add OfferAcceptedEmail

Add ability to cc emails

camerojo committed:
Disable Opt in/out functionality by commenting out relevant user interface buttons etc.

camerojo committed:
Change 'ng serve' to 'npm start' so that we get the --host Angular debug fix

camerojo committed:
Fix bug with job accepted trigger

camerojo committed:
Remove copyrights from html mail templates

camerojo committed:
Refer to Pathway Club welcome page

camerojo committed:
Change sendOfferAccepted to take a candidate, and to pass in the candidate number.

Display candidate number in job accepted emails.

camerojo committed:
Add punctuation

camerojo committed:
More punctuation

sadatmalik commented on server/src/main/resources/mail/job-accepted.html:
```diff
@@ -0,0 +1,47 @@
+  <p>
+    Members of Pathway Club look forward to seeing you on our Slack where they can welcome you
+    personally and help you where they can.
+  </p>
```
Love this email!
